### PR TITLE
Add authorization

### PR DESCRIPTION
**Pull request by smunoz1988**
Hello!

for this milestone, We made this changes:

- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- Implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- Implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

Note: style was not required. We focus on the functionality.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  def initialize(user)
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, Post, author_id: user.id
+      can :create, Comment, author_id: user.id
+      can :create, Like
+      can :destroy, Post do |post|
```
Comment (TadesseDev) [inline app/models/ability.rb:12]: - [Optional] dear @smunoz1988, to simplify the matter and make your authorization rules a bit more clear, consider using `can :manage` to allow all action to a user. 
> Something like 
```ruby
can :manage, Post, author: user # This is a more efficient way of giving all the permission. 
```

**Review by TadesseDev — APPROVED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] You have written a clean code ✅
 - [x] Functrional app. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — COMMENTED** [review]
Hi @smunoz1988 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Invalid Code Review Request 🔴 

You have submitted a project, that has been merged. Kindly send an open PR

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by smunoz1988** [comment]
> Hi @smunoz1988 ,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> # Invalid Code Review Request 🔴
> 
> You have submitted a project, that has been merged. Kindly send an open PR
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

hello @Shedrack-Sunday 

I don't understand. The pull request was already approved. you can check on @TadesseDev  comment above.

**Review by rotimiazeez — COMMENTED** [review]
Hi TEAM 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by TadesseDev** [comment]
Sorry for the inconvenience Team, there was a blackout after my review before the submission. which prevents me from submitting the approval.

**Pull request merged**